### DISCUSSION TITLE
Help Building 32bit pyrfc

### DISCUSSION THREAD

**Discussion by josephkirby** [SAP/PyRFC#357] [Q&A]
I need help building a 32bit version of pyrfc.  I have the latest version of Build Tools installed.  I have python 32bit installed.  And I have 32bit SAP GUI 8.0 installed.  What do I need to do to build pyrfc?

**Comment by bsrdjan** [comment]
Here are [Requirements](https://github.com/SAP/PyRFC?tab=readme-ov-file#requirements)

**Reply by josephkirby** [reply]
That doesn't answer the question.  What are the steps to build the 32bit version?

**Reply by bsrdjan** [reply]
Steps are the same as for 64 bit version, only the 32 bit Python and 32 bit SAP NW RFC SDK shall be used.

The 32 bit pyrfc can be built on 64 or 32 bit operating system, supported by SAP NW RFC SDK (see https://me.sap.com/notes/2573790).

Compiler options are in same SAP OSS note where 64 bit options are provided https://me.sap.com/notes/2573953

**Reply by josephkirby** [reply]
I am getting this error now:
![image](https://github.com/SAP/PyRFC/assets/84047648/cf3cd580-620b-44d7-94b8-39e39c4f5c8d)

**Reply by josephkirby** [reply]
I got that fixed and now I'm getting a whole bunch of errors.  Not sure what to do.

`C:\git\PyRFC>C:\Users\jkirby\AppData\Local\Programs\Python\Python311-32\python.exe setup.py build
C:\Users\jkirby\AppData\Local\Programs\Python\Python311-32\Lib\site-packages\setuptools\config\pyprojecttoml.py:108: _BetaConfiguration: Support for `[tool.setuptools]` in `pyproject.toml` is still *beta*.
  warnings.warn(msg, _BetaConfiguration)
running build
running build_py
running build_ext
building 'pyrfc._cyrfc' extension
"C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\bin\HostX86\x86\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -DSAPonNT -D_CRT_NON_CONFORMING_SWPRINTFS -D_CRT_SECURE_NO_DEPRECATES -D_CRT_NONSTDC_NO_DEPRECATE -D_AFXDLL -DWIN32 -D_WIN32_WINNT=0x0502 -DWIN64 -D_AMD64_ -DNDEBUG -DSAPwithUNICODE -DUNICODE -D_UNICODE -DSAPwithTHREADS -D_ATL_ALLOW_CHAR_UNSIGNED -D_LARGEFILE_SOURCE -D_CONSOLE -DSAP_PLATFORM_MAKENAME=ntintel -IC:\Users\jkirby\AppData\Local\Programs\Python\Python311-32\include -IC:\Users\jkirby\AppData\Local\Programs\Python\Python311-32\Include "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt" /EHsc /Tpsrc/pyrfc/_cyrfc.cpp /Fobuild\temp.win32-cpython-311\Release\src/pyrfc/_cyrfc.obj -IC:\Windows\SysWOW64 -IC:\Users\jkirby\AppData\Local\Programs\Python\Python311-32\Lib\inspect.py\Include -IC:\Users\jkirby\AppData\Local\Programs\Python\Python311-32\Lib\inspect.py\Include\PC /EHs /GL /Gy /J /MD /nologo /O2 /Oy- /we4552 /we4700 /we4789 /W3 /Z7
_cyrfc.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\sal.h(2361): error C2144: syntax error: 'int' should be preceded by ';'
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include\sal.h(2361): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
src/pyrfc/_cyrfc.cpp(1643): error C3646: '_server_handle': unknown override specifier
src/pyrfc/_cyrfc.cpp(1643): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
src/pyrfc/_cyrfc.cpp(1658): error C3646: '_throughput_handle': unknown override specifier
src/pyrfc/_cyrfc.cpp(1658): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
src/pyrfc/_cyrfc.cpp(2765): error C2065: 'SAP_ULLONG': undeclared identifier
src/pyrfc/_cyrfc.cpp(2765): error C2146: syntax error: missing ')' before identifier 'value'
src/pyrfc/_cyrfc.cpp(8026): error C3861: 'RfcLanguageIsoToSap': identifier not found
src/pyrfc/_cyrfc.cpp(8242): error C3861: 'RfcLanguageSapToIso': identifier not found
src/pyrfc/_cyrfc.cpp(8568): error C3861: 'RfcLoadCryptoLibrary': identifier not found
src/pyrfc/_cyrfc.cpp(8894): error C3861: 'RfcCancel': identifier not found
src/pyrfc/_cyrfc.cpp(26316): error C2039: 'isStateful': is not a member of '_RFC_SERVER_CONTEXT'
C:\Users\jkirby\AppData\Local\Programs\Python\Python311-32\include\sapnwrfc.h(337): note: see declaration of '_RFC_SERVER_CONTEXT'
src/pyrfc/_cyrfc.cpp(28544): error C2039: '_server_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(1635): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(28546): error C2039: '_server_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(1635): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(28618): error C2039: '_server_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(1635): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(29620): error C2039: '_server_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(1635): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(29620): error C3861: 'RfcCreateServer': identifier not found
src/pyrfc/_cyrfc.cpp(29659): error C2039: '_server_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(1635): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(32460): error C2664: 'RFC_RC RfcInstallBgRfcHandlers(const SAP_UC *,RFC_ON_CHECK_UNIT,RFC_ON_COMMIT_UNIT,RFC_ON_ROLLBACK_UNIT,RFC_ON_CONFIRM_UNIT,RFC_ON_GET_UNIT_STATE,RFC_ERROR_INFO *)': cannot convert argument 2 from 'RFC_RC (__cdecl *)(RFC_CONNECTION_HANDLE,const RFC_UNIT_IDENTIFIER *)' to 'RFC_ON_CHECK_UNIT'
src/pyrfc/_cyrfc.cpp(32460): note: None of the functions with this name in scope match the target type
C:\Users\jkirby\AppData\Local\Programs\Python\Python311-32\include\sapnwrfc.h(1566): note: see declaration of 'RfcInstallBgRfcHandlers'
src/pyrfc/_cyrfc.cpp(32460): note: while trying to match the argument list '(SAP_UC *, overloaded-function, overloaded-function, overloaded-function, overloaded-function, overloaded-function, RFC_ERROR_INFO *)'
src/pyrfc/_cyrfc.cpp(33273): error C2664: 'RFC_RC RfcInstallGenericServerFunction(RFC_SERVER_FUNCTION,RFC_FUNC_DESC_CALLBACK,RFC_ERROR_INFO *)': cannot convert argument 1 from 'RFC_RC (__cdecl *)(RFC_CONNECTION_HANDLE,RFC_FUNCTION_HANDLE,RFC_ERROR_INFO *)' to 'RFC_SERVER_FUNCTION'
src/pyrfc/_cyrfc.cpp(33273): note: None of the functions with this name in scope match the target type
C:\Users\jkirby\AppData\Local\Programs\Python\Python311-32\include\sapnwrfc.h(1480): note: see declaration of 'RfcInstallGenericServerFunction'
src/pyrfc/_cyrfc.cpp(33273): note: while trying to match the argument list '(overloaded-function, overloaded-function, RFC_ERROR_INFO *)'
src/pyrfc/_cyrfc.cpp(33322): error C2039: '_server_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(1635): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(33322): error C3861: 'RfcLaunchServer': identifier not found
src/pyrfc/_cyrfc.cpp(33768): error C2039: '_server_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(1635): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(33768): error C3861: 'RfcShutdownServer': identifier not found
src/pyrfc/_cyrfc.cpp(33943): error C2039: '_server_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(1635): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(34034): error C2039: '_server_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(1635): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(34034): error C3861: 'RfcDestroyServer': identifier not found
src/pyrfc/_cyrfc.cpp(34063): error C2039: '_server_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(1635): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(34355): error C2065: 'RFC_SERVER_ATTRIBUTES': undeclared identifier
src/pyrfc/_cyrfc.cpp(34355): error C2146: syntax error: missing ';' before identifier '__pyx_v_attributes'
src/pyrfc/_cyrfc.cpp(34355): error C2065: '__pyx_v_attributes': undeclared identifier
src/pyrfc/_cyrfc.cpp(34380): error C2039: '_server_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(1635): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Server'
src/pyrfc/_cyrfc.cpp(34380): error C2065: '__pyx_v_attributes': undeclared identifier
src/pyrfc/_cyrfc.cpp(34380): error C3861: 'RfcGetServerAttributes': identifier not found
src/pyrfc/_cyrfc.cpp(34432): error C2065: '__pyx_v_attributes': undeclared identifier
src/pyrfc/_cyrfc.cpp(34432): error C3861: 'RfcGetServerStateAsString': identifier not found
src/pyrfc/_cyrfc.cpp(34454): error C2065: '__pyx_v_attributes': undeclared identifier
src/pyrfc/_cyrfc.cpp(34454): error C2065: 'RFC_MULTI_COUNT_REGISTERED_SERVER': undeclared identifier
src/pyrfc/_cyrfc.cpp(34538): error C2065: '__pyx_v_attributes': undeclared identifier
src/pyrfc/_cyrfc.cpp(34559): error C2065: '__pyx_v_attributes': undeclared identifier
src/pyrfc/_cyrfc.cpp(34580): error C2065: '__pyx_v_attributes': undeclared identifier
src/pyrfc/_cyrfc.cpp(34592): error C2065: '__pyx_v_attributes': undeclared identifier
src/pyrfc/_cyrfc.cpp(36505): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(36529): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(36529): error C3861: 'RfcCreateThroughput': identifier not found
src/pyrfc/_cyrfc.cpp(36905): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(37062): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(37062): error C3861: 'RfcSetThroughputOnConnection': identifier not found
src/pyrfc/_cyrfc.cpp(37264): error C2065: 'RFC_THROUGHPUT_HANDLE': undeclared identifier
src/pyrfc/_cyrfc.cpp(37264): error C2146: syntax error: missing ';' before identifier '__pyx_v_throughput'
src/pyrfc/_cyrfc.cpp(37264): error C2065: '__pyx_v_throughput': undeclared identifier
src/pyrfc/_cyrfc.cpp(37287): error C2065: '__pyx_v_throughput': undeclared identifier
src/pyrfc/_cyrfc.cpp(37287): error C3861: 'RfcGetThroughputFromConnection': identifier not found
src/pyrfc/_cyrfc.cpp(37395): error C2065: '__pyx_v_throughput': undeclared identifier
src/pyrfc/_cyrfc.cpp(37602): error C3861: 'RfcRemoveThroughputFromConnection': identifier not found
src/pyrfc/_cyrfc.cpp(37761): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(37761): error C3861: 'RfcResetThroughput': identifier not found
src/pyrfc/_cyrfc.cpp(37895): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(37905): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(37905): error C3861: 'RfcDestroyThroughput': identifier not found
src/pyrfc/_cyrfc.cpp(37914): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(38327): error C2065: 'SAP_ULLONG': undeclared identifier
src/pyrfc/_cyrfc.cpp(38327): error C2146: syntax error: missing ';' before identifier '__pyx_v_numberOfCalls'
src/pyrfc/_cyrfc.cpp(38327): error C2065: '__pyx_v_numberOfCalls': undeclared identifier
src/pyrfc/_cyrfc.cpp(38328): error C2065: 'SAP_ULLONG': undeclared identifier
src/pyrfc/_cyrfc.cpp(38328): error C2146: syntax error: missing ';' before identifier '__pyx_v_sentBytes'
src/pyrfc/_cyrfc.cpp(38328): error C2065: '__pyx_v_sentBytes': undeclared identifier
src/pyrfc/_cyrfc.cpp(38329): error C2065: 'SAP_ULLONG': undeclared identifier
src/pyrfc/_cyrfc.cpp(38329): error C2146: syntax error: missing ';' before identifier '__pyx_v_receivedBytes'
src/pyrfc/_cyrfc.cpp(38329): error C2065: '__pyx_v_receivedBytes': undeclared identifier
src/pyrfc/_cyrfc.cpp(38330): error C2065: 'SAP_ULLONG': undeclared identifier
src/pyrfc/_cyrfc.cpp(38330): error C2146: syntax error: missing ';' before identifier '__pyx_v_applicationTime'
src/pyrfc/_cyrfc.cpp(38330): error C2065: '__pyx_v_applicationTime': undeclared identifier
src/pyrfc/_cyrfc.cpp(38331): error C2065: 'SAP_ULLONG': undeclared identifier
src/pyrfc/_cyrfc.cpp(38331): error C2146: syntax error: missing ';' before identifier '__pyx_v_totalTime'
src/pyrfc/_cyrfc.cpp(38331): error C2065: '__pyx_v_totalTime': undeclared identifier
src/pyrfc/_cyrfc.cpp(38332): error C2065: 'SAP_ULLONG': undeclared identifier
src/pyrfc/_cyrfc.cpp(38332): error C2146: syntax error: missing ';' before identifier '__pyx_v_serializationTime'
src/pyrfc/_cyrfc.cpp(38332): error C2065: '__pyx_v_serializationTime': undeclared identifier
src/pyrfc/_cyrfc.cpp(38333): error C2065: 'SAP_ULLONG': undeclared identifier
src/pyrfc/_cyrfc.cpp(38333): error C2146: syntax error: missing ';' before identifier '__pyx_v_deserializationTime'
src/pyrfc/_cyrfc.cpp(38333): error C2065: '__pyx_v_deserializationTime': undeclared identifier
src/pyrfc/_cyrfc.cpp(38366): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(38420): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(38420): error C2065: '__pyx_v_numberOfCalls': undeclared identifier
src/pyrfc/_cyrfc.cpp(38420): error C3861: 'RfcGetNumberOfCalls': identifier not found
src/pyrfc/_cyrfc.cpp(38461): error C2065: '__pyx_v_numberOfCalls': undeclared identifier
src/pyrfc/_cyrfc.cpp(38461): error C3861: '__Pyx_PyInt_From_SAP_ULLONG': identifier not found
src/pyrfc/_cyrfc.cpp(38473): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(38473): error C2065: '__pyx_v_sentBytes': undeclared identifier
src/pyrfc/_cyrfc.cpp(38473): error C3861: 'RfcGetSentBytes': identifier not found
src/pyrfc/_cyrfc.cpp(38514): error C2065: '__pyx_v_sentBytes': undeclared identifier
src/pyrfc/_cyrfc.cpp(38514): error C3861: '__Pyx_PyInt_From_SAP_ULLONG': identifier not found
src/pyrfc/_cyrfc.cpp(38526): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(38526): error C2065: '__pyx_v_receivedBytes': undeclared identifier
src/pyrfc/_cyrfc.cpp(38526): error C3861: 'RfcGetReceivedBytes': identifier not found
src/pyrfc/_cyrfc.cpp(38567): error C2065: '__pyx_v_receivedBytes': undeclared identifier
src/pyrfc/_cyrfc.cpp(38567): error C3861: '__Pyx_PyInt_From_SAP_ULLONG': identifier not found
src/pyrfc/_cyrfc.cpp(38579): error C2039: '_throughput_handle': is not a member of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(1655): note: see declaration of '__pyx_obj_5pyrfc_6_cyrfc_Throughput'
src/pyrfc/_cyrfc.cpp(38579): error C2065: '__pyx_v_applicationTime': undeclared identifier
src/pyrfc/_cyrfc.cpp(38579): error C3861: 'RfcGetApplicationTime': identifier not found
src/pyrfc/_cyrfc.cpp(38620): error C2065: '__pyx_v_applicationTime': undeclared identifier
src/pyrfc/_cyrfc.cpp(38620): error C3861: '__Pyx_PyInt_From_SAP_ULLONG': identifier not found
src/pyrfc/_cyrfc.cpp(38620): fatal error C1003: error count exceeds 100; stopping compilation
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.38.33130\\bin\\HostX86\\x86\\cl.exe' failed with exit code 2`

Could someone build the 32bit wheel for me?

**Comment by bsrdjan** [comment]
PyRFC 32bit version is not maintained any more and build options need to be investigated. If build support needed, please create customer ticket for `BC-MID-RFC-SDK` component and refer to this GH issue.

The build command, as described in README, might help: https://github.com/SAP/PyRFC?tab=readme-ov-file#download-and-installation. In case of issues, delete cpp files from source folder and re-test

`python -m build --wheel --outdir dist`

**Comment by josephkirby** [comment]
I'm making some progress but still getting errors.  Can you assist?

`src/pyrfc/_cyrfc.cpp(32404): error C2664: 'RFC_RC RfcInstallBgRfcHandlers(const SAP_UC *,RFC_ON_CHECK_UNIT,RFC_ON_COMMIT_UNIT,RFC_ON_ROLLBACK_UNIT,RFC_ON_CONFIRM_UNIT,RFC_ON_GET_UNIT_STATE,RFC_ERROR_INFO *)': cannot convert argument 2 from 'RFC_RC (__cdecl *)(RFC_CONNECTION_HANDLE,const RFC_UNIT_IDENTIFIER *)' to 'RFC_ON_CHECK_UNIT'
src/pyrfc/_cyrfc.cpp(32404): note: None of the functions with this name in scope match the target type
C:\Users\jkirby\AppData\Local\Programs\Python\Python311-32\include\sapnwrfc.h(2446): note: see declaration of 'RfcInstallBgRfcHandlers'
src/pyrfc/_cyrfc.cpp(32404): note: while trying to match the argument list '(SAP_UC *, overloaded-function, overloaded-function, overloaded-function, overloaded-function, overloaded-function, RFC_ERROR_INFO *)'
src/pyrfc/_cyrfc.cpp(33217): error C2664: 'RFC_RC RfcInstallGenericServerFunction(RFC_SERVER_FUNCTION,RFC_FUNC_DESC_CALLBACK,RFC_ERROR_INFO *)': cannot convert argument 1 from 'RFC_RC (__cdecl *)(RFC_CONNECTION_HANDLE,RFC_FUNCTION_HANDLE,RFC_ERROR_INFO *)' to 'RFC_SERVER_FUNCTION'
src/pyrfc/_cyrfc.cpp(33217): note: None of the functions with this name in scope match the target type
C:\Users\jkirby\AppData\Local\Programs\Python\Python311-32\include\sapnwrfc.h(2360): note: see declaration of 'RfcInstallGenericServerFunction'
src/pyrfc/_cyrfc.cpp(33217): note: while trying to match the argument list '(overloaded-function, overloaded-function, RFC_ERROR_INFO *)'
src/pyrfc/_cyrfc.cpp(45958): warning C4018: '<': signed/unsigned mismatch
src/pyrfc/_cyrfc.cpp(60036): warning C4551: function call missing argument list
src/pyrfc/_cyrfc.cpp(60637): warning C4551: function call missing argument list
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.38.33130\\bin\\HostX86\\x86\\cl.exe' failed with exit code 2

ERROR Backend subprocess exited when trying to invoke build_wheel`
